tests/BuildScriptGenerator.Tests/Node/NodePlatformTest.cs

```diff
@@ -160,6 +161,122 @@ public void BuildScript_DoesNotHaveSdkInstallScript_IfDynamicInstallNotEnabled_A
+                commonOptions,
+                new NodeScriptGeneratorOptions(),
+                new NodePlatformInstaller(Options.Create(commonOptions), new TestEnvironment()));
+            var repo = new MemorySourceRepo();
+            repo.AddFile(packageJson, NodeConstants.PackageJsonFileName);
+            var context = CreateContext(repo);
+            context.NodeVersion = "10.10";
+            context.Properties[NodePlatform.RequireBuildPropertyKey] = "true";
+
+            // Act
+            var buildScriptSnippet = nodePlatform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(buildScriptSnippet);
+            Assert.Contains("npm run build:azure", buildScriptSnippet.BashBuildScriptSnippet);
+        }
+
+        [Fact]
+        public void GeneratedBuildSnippet_DoesNotThrowException_IfCustomRunCommandIsProvided_AndRequireBuildPropertyIsSet()
+        {
+            // Arrange
+            const string packageJson = @"{
+              ""main"": ""server.js"",
+              ""scripts"": {
+              },
+            }";
+            var commonOptions = new BuildScriptGeneratorOptions();
+            var nodePlatform = CreateNodePlatform(
+                commonOptions,
+                new NodeScriptGeneratorOptions { CustomRunBuildCommand = "custom command here" },
+                new NodePlatformInstaller(Options.Create(commonOptions), new TestEnvironment()));
+            var repo = new MemorySourceRepo();
+            repo.AddFile(packageJson, NodeConstants.PackageJsonFileName);
+            var context = CreateContext(repo);
+            context.NodeVersion = "10.10";
+            context.Properties[NodePlatform.RequireBuildPropertyKey] = "true";
+
+            // Act
+            var buildScriptSnippet = nodePlatform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(buildScriptSnippet);
+            Assert.Contains("custom command here", buildScriptSnippet.BashBuildScriptSnippet);
+        }
+
+        [Fact]
+        public void GeneratedBuildSnippet_ThrowsException_IfRequireBuildPropertyIsSet_AndNoBuildStepIsProvided()
+        {
+            // Arrange
+            const string packageJson = @"{
+              ""main"": ""server.js"",
+              ""scripts"": {
+              },
+            }";
+            var commonOptions = new BuildScriptGeneratorOptions();
+            var nodePlatform = CreateNodePlatform(
+                commonOptions,
+                new NodeScriptGeneratorOptions(),
+                new NodePlatformInstaller(Options.Create(commonOptions), new TestEnvironment()));
+            var repo = new MemorySourceRepo();
+            repo.AddFile(packageJson, NodeConstants.PackageJsonFileName);
+            var context = CreateContext(repo);
+            context.NodeVersion = "10.10";
+            context.Properties[NodePlatform.RequireBuildPropertyKey] = "true";
```

Comment:
nit: can we add  some negative tests as well

e.g.
context.Properties[NodePlatform.RequireBuildPropertyKey] = ""
context.Properties[NodePlatform.RequireBuildPropertyKey]="false"

Reply:
Sure, sounds good.